watch.mjs

```diff
@@ -0,0 +1,42 @@
+import fsCallback from 'node:fs';
+import { buildComponentSets, listComponentSetPaths, validateComponentSet } from './build.mjs';
+
+export async function watchComponentSets() {
+    // Watch does not rebuild vendor.js and design.css
+    await buildComponentSets();
+
+    for (const componentSetPath of await listComponentSetPaths()) {
+        await watchComponentSet(componentSetPath);
+    }
+}
+
+export async function watchComponentSet(componentSetPath) {
+    let dirty = false;
+    let activePromise = null;
+
+    const doValidateComponentSet = () => {
+        dirty = false;
+        activePromise = validateComponentSet(componentSetPath)
+            .catch((err) => {
+                console.error(err);
+            })
+            .finally(() => {
+                activePromise = undefined;
+
+                if (dirty) {
+                    doValidateComponentSet();
+                }
+            });
+    };
+
+    fsCallback.watch(componentSetPath, { recursive: true }, (eventType, filename) => {
+        console.log(`\nThe file "${filename}" was modified!`);
+        console.log('The type of change was:', eventType);
+
+        if (activePromise) {
```

Comment:
Small issue: When deleting a component set folder in watch mode,  you get an error:
```
[Error: ENOENT: no such file or directory, scandir 'component-sets/default-components copy'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: 'component-sets/default-components copy'
}
```